--- a/lld/COFF/DLL.cpp
+++ b/lld/COFF/DLL.cpp
@@ -386,6 +386,20 @@
   Defined *helper = nullptr;
 };
 
+class TailMergePDataChunkARM64 : public NonSectionChunk {
+public:
+  explicit TailMergePDataChunkARM64(Chunk *tm) : tm(tm) { setAlignment(4); }
+  size_t getSize() const override { return 2 * sizeof(uint32_t); }
+  MachineTypes getMachine() const override { return ARM64; }
+  void writeTo(uint8_t *buf) const override {
+    write32le(buf, tm->getRVA());
+    write32le(buf + 4, 1 | ((tm->getSize() / 4) << 2) | (3 << 21) |
+                          ((224 / 16) << 23));
+  }
+
+private:
+  Chunk *tm;
+};
 class TailMergePDataChunkX64 : public NonSectionChunk {
 public:
   TailMergePDataChunkX64(Chunk *tm, Chunk *unwind) : tm(tm), unwind(unwind) {
@@ -1062,6 +1076,8 @@
     if (!symtab.tailMergeUnwindInfoChunk)
       symtab.tailMergeUnwindInfoChunk = make<TailMergeUnwindInfoX64>();
     return make<TailMergePDataChunkX64>(tm, symtab.tailMergeUnwindInfoChunk);
+  case ARM64:
+    return make<TailMergePDataChunkARM64>(tm);
     // FIXME: Add support for other architectures.
   default:
     return nullptr; // Just don't generate unwind info.
